Azure/azure-cli
# [Storage] Fix #16917: az storage account generate-sas fails if a connection string is provided

**Description**

Fix #16917: az storage account generate-sas fails if a connection string is provided

In PR #16034, a validator is added to check `accout_name` not `None` or `''`. However, `account_name` can be parsed from `connection_string` in another validator for `connecting_string`, so when a valid `connection_string` is provided, `account_name` is not necessary.

This PR fixed this issue by checking `account_name` and `account_key` before invoking sdk function, at which time `account_name` has been parsed properly through `connection_string` validator.

**Testing Guide**

Expected failure cases:
`az storage account generate-sas --account-name ""`
`az storage account generate-sas --connection-string $invalid_connection_string`
Success cases:
`az storage account generate-sas --account-name mystorageaccount`
`az storage account generate-sas --connection-string $valid_connection_string`

---

This checklist is used to make sure that common guidelines for a pull request are followed.
